dumux-precice/couplingadapter.cc

```diff
@@ -28,102 +24,73 @@ CouplingAdapter &CouplingAdapter::getInstance()
 }
 
 void CouplingAdapter::announceSolver(const std::string &name,
-                                     const std::string configurationFileName,
+                                     const std::string &configurationFileName,
                                      const int rank,
                                      const int size)
 {
     assert(precice_ == nullptr);
-    precice_ = std::make_unique<precice::SolverInterface>(
+    precice_ = std::make_unique<precice::Participant>(
         name, configurationFileName, rank, size);
     wasCreated_ = true;
 }
 
-size_t CouplingAdapter::announceQuantity(const std::string &name,
-                                         const QuantityType quantity_type)
+void CouplingAdapter::announceQuantity(const precice::string_view &meshName,
+                                       const precice::string_view &dataName)
 {
     assert(meshWasCreated_);
-    auto it = std::find(dataNames_.begin(), dataNames_.end(), name);
-    if (it != dataNames_.end()) {
+    const std::string key = createKeyFromName(meshName, dataName);
+    if (dataMap_.find(key) != dataMap_.end()) {
         throw(std::runtime_error(" Error! Duplicate quantity announced! "));
     }
-    dataNames_.push_back(name);
-    preciceDataID_.push_back(precice_->getDataID(name, meshID_));
-    const int quantity_dimension =
-        (quantity_type == QuantityType::Scalar) ? 1 : getDimensions();
-    dataVectors_.push_back(
-        std::vector<double>(vertexIDs_.size() * quantity_dimension));
 
-    return getNumberOfQuantities() - 1;
+    int dataDimension = precice_->getDataDimensions(meshName, dataName);
+    std::vector<double> dataValues(vertexIDs_.size() * dataDimension);
+    dataMap_.insert(std::make_pair(key, dataValues));
 }
 
-size_t CouplingAdapter::announceScalarQuantity(const std::string &name)
-{
-    return announceQuantity(name, QuantityType::Scalar);
-}
-
-size_t CouplingAdapter::announceVectorQuantity(const std::string &name)
-{
-    return announceQuantity(name, QuantityType::Vector);
-}
```

Comment:
Do we not need these `announce*Quantity` functions at all now?

Reply:
The `announceQuantity` function can announce both scalar and vector quantity now:  the data dimension would be detected in that function. 
So I removed these two.